lib/engine/game/g_1849/step/buy_token.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class BuyToken < Engine::Step::Base
+          ACTIONS = %w[buy_token pass].freeze
+          MIN_PRICE = 1
+
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] unless can_buy_token?(entity)
+
+            ACTIONS
+          end
+
+          def round_state
+            super.merge(
+              {
+                bought_token: false,
+              }
+            )
+          end
+
+          def setup
+            super
+            @round.bought_token = false
+          end
+
+          def can_buy_token?(entity)
+            @game.acquiring_station_tokens? &&
+              @game.phase.tiles.include?(:green) &&
+              current_entity == entity &&
+              !@round.bought_token &&
+              !available_tokens(entity).empty? &&
+              MIN_PRICE <= buying_power(entity)
+          end
+
+          def available_tokens(entity)
+            entity.tokens_by_type
+          end
+
+          def can_sell_token?(token, entity)
+            token.corporation.placed_tokens.size > 1 &&
```

Comment:
This check in unnecessary if you can't sell your first token. By definition you have more than one placed token (your home + this one).